cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -239,66 +244,83 @@ public KafkaBrokerConfigurationBuilder withListeners(
         List<String> advertisedListeners = new ArrayList<>();
         List<String> securityProtocol = new ArrayList<>();
 
-        // Control Plane listener
-        listeners.add(CONTROL_PLANE_LISTENER_NAME + "://0.0.0.0:9090");
-        if (!useKRaft) {
-            advertisedListeners.add(String.format("%s://%s:9090",
-                    CONTROL_PLANE_LISTENER_NAME,
+        boolean isKraftControllerOnly = useKRaft && node.controller() && !node.broker();
+        boolean isKraftBrokerOnly = useKRaft && node.broker() && !node.controller();
+
+        // Control Plane listener => ZooKeeper based brokers and KRaft controllers (including mixed nodes). But not on KRaft broker only nodes
+        if (!isKraftBrokerOnly) {
+            listeners.add(CONTROL_PLANE_LISTENER_NAME + "://0.0.0.0:9090");
+
+            if (!useKRaft) {
+                advertisedListeners.add(String.format("%s://%s:9090",
+                        CONTROL_PLANE_LISTENER_NAME,
+                        // Pod name constructed to be templatable for each individual ordinal
+                        DnsNameGenerator.podDnsNameWithoutClusterDomain(namespace, KafkaResources.brokersServiceName(clusterName), node.podName())
+                ));
+            }
+
+            securityProtocol.add(CONTROL_PLANE_LISTENER_NAME + ":SSL");
+            configureControlPlaneListener();
+        } else {
+            // But even KRaft broker only node needs it in the security protocol map and the security configuration
+            configureControlPlaneListener();
+            securityProtocol.add(CONTROL_PLANE_LISTENER_NAME + ":SSL");
+        }
```

Comment:
two questions:

1. any reasons why the two lines of code above are in a different order in the two `if-else` branches?
2. can we avoid the `else` and having the above two lines out of it because it's anyway in common with the `if` branch?